core/internal/stream/sender.go

```diff
@@ -1095,8 +1095,8 @@ func (s *Sender) sendRun(record *spb.Record, run *spb.RunRecord) {
 	s.startState = runbranch.NewRunParams(run, s.settings)
 
 	isResume := s.settings.GetResume()
-	isRewind := run.GetBranchPoint() != nil && run.GetBranchPoint().Type == spb.BranchPointRecord_BRANCH_POINT_REWIND
-	isFork := run.GetBranchPoint() != nil && run.GetBranchPoint().Type == spb.BranchPointRecord_BRANCH_POINT_FORK
+	isRewind := run.GetBranchPoint().GetRun() == run.RunId
```

Comment:
maybe a comment about the implicit assumption that when branchpoint not set it is an empty string,

Reply:
it's a protobuf convention and why the `Get` methods exist (`run.BranchPoint.Run` would panic because `BranchPoint` is nil), so a comment would be redundant IMO

Reply:
yeah i know how the gets work in protobuf it is just implicit logic that we both check that this field exists and the value is correct, but if you feel it is clear from the context can skip it